src/main/java/com/mraof/minestuck/data/BetterCombatProvider.java

```diff
@@ -0,0 +1,183 @@
+{
+	private final PackOutput output;
+	
+	Map<String, String> weaponWithParent = new HashMap<>();
+	
+	public BetterCombatProvider(PackOutput output)
+	{
+		this.output = output;
+	}
+	
+	/**
+	 * First gets all the weapons from MSItems and adds it to a Map where the key is the registry name and the value is the data.
+	 * Then it checks for custom defined weapons to use either in addition to the existing entries or to replace
+	 */
+	@Override
+	public final CompletableFuture<?> run(CachedOutput cache)
+	{
+		List<CompletableFuture<?>> futures = new ArrayList<>();
+		Path basePath = this.output.getOutputFolder(PackOutput.Target.DATA_PACK).resolve(Minestuck.MOD_ID).resolve("weapon_attributes");
+		
+		MSItems.REGISTER.getEntries().forEach(weaponHolder ->
+		{
+			if(weaponHolder.asOptional().isEmpty())
+				return;
+			
+			if(weaponHolder.get() instanceof WeaponItem weaponItem && weaponItem.getToolType() != null)
+			{
+				String parent = getWeaponParent(weaponItem.getToolType());
+				
+				if(!parent.isEmpty())
+					weaponWithParent.put(weaponHolder.getKey().location().getPath(), parent);
+			}
+		});
+		
+		customWeapons();
+		
+		for(Map.Entry<String, String> entry : weaponWithParent.entrySet())
+		{
+			JsonObject object = new JsonObject();
+			
+			object.addProperty("parent", entry.getValue());
+			
+			futures.add(DataProvider.saveStable(cache, object, basePath.resolve(entry.getKey() + ".json")));
+		}
+		
+		return CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+	}
+	
+	public String getWeaponParent(MSToolType toolType)
+	{
+		if(toolType == SICKLE_TOOL)
+			return "bettercombat:sickle";
+		else if(toolType == SCYTHE_TOOL)
+			return "bettercombat:scythe";
+		else if(toolType == CLAWS_TOOL)
+			return "bettercombat:claw";
+		else if(toolType == PICKAXE_TOOL)
+			return "bettercombat:pickaxe";
+		else if(toolType == HAMMER_TOOL)
+			return "bettercombat:hammer";
+		else if(toolType == AXE_TOOL)
+			return "bettercombat:axe";
+		else if(toolType == CHAINSAW_TOOL)
+			return "minestuck:chainsaw_base";
+		else if(toolType == CLUB_TOOL || toolType == SHOVEL_TOOL)
+			return "bettercombat:mace";
+		else if(toolType == SWORD_TOOL || toolType == KEY_TOOL || toolType == BATON_TOOL)
+			return "bettercombat:sword";
+		else if(toolType == KNIFE_TOOL || toolType == FAN_TOOL)
+			return "bettercombat:dagger";
+		else if(toolType == LANCE_TOOL)
+			return "bettercombat:lance";
+		else if(toolType == STAFF_TOOL)
+			return "bettercombat:battlestaff";
+		else if(toolType == CANE_TOOL)
+			return "minestuck:cane_base";
+		else if(toolType == FORK_TOOL)
+			return "minestuck:fork_base";
+		else if(toolType == WAND_TOOL)
+			return "bettercombat:wand";
+		return "";
+	}
+	
+	public void customWeapons()
+	{
+		addWeapon(MSItems.MAILBOX, "bettercombat:hammer");
+		addWeapon(MSItems.ESTROGEN_EMPOWERED_EVERYTHING_ERADICATOR, "bettercombat:hammer");
+		addWeapon(MSItems.BOOMBOX_BEATER, "bettercombat:hammer");
```

Comment:
Because these string literals must match the name of the parent attribute exactly, how about putting the string literals for all the different parents into string constants, and then use the constants here?